report/3.Theory/Dependability.tex

```diff
@@ -0,0 +1,45 @@
+\section{Dependability}
+Dependability in a real-time system is divided into two fields: hardware and software. 
+In the following subsections, the different methods and techniques used to achieve dependability will be elaborated upon.  
+In order to understand and achieve dependability, a fundamental understanding in regards to faults is needed, which will therefore be covered in this section.
+
+When dealing with dependability of a system, a key observation to be made is what defines the system. 
+\begin{quote}
+\textbf{System}: A system is an entity that interacts with other entities, i.e., other systems, including hardware, software, humans, and the physical world with its natural phenomena.
+These systems are the environment of the given system. 
+- John Knight \cite{FundementalsOfDependableComputingForSoftwareEngineers}
+\end{quote}
+It is said that a system is a system composed of systems, that are composed of systems and so on. 
+However, there is a system boundary between all the systems that composes the system and the system itself. 
+Hence dividing the system into a subject system and the systems in the environment. 
+This is important to notice when evaluating the dependability of the system. 
+It is inherent that a system, which is composed of an independent system, can never be more dependent than that system. 
+
+Whenever a system does not provided the service specified by its requirements, it is said that the system has a service failure. 
+Once a service failure is observed as an direct result of an error with was activated by a fault. 
+
+\subsection{Faults}
+There are broadly speaking different kinds of faults: degradation faults, design faults and byzantine faults. 
+An degradation fault is a result of the natural wear and tear of the hardware components.
+A design fault is, as the name implies, an fault that is caused by an error or poor decision made during the design phase of the component. 
+A byzantine fault is when a component of the system, or a system in its entirety, behaves in an unexpected way. 
+Byzantine fault are the hardest to detect and correct since their behavior is by definition unpredictable. 
```

Comment:
Faults no? One fault, more faults.
```suggestion
Byzantine faults are the hardest to detect and correct since their behavior is by definition unpredictable. 
```